docs/design/2020-08-04-global-index.md

```diff
@@ -0,0 +1,191 @@
+
+## Background
+
+### Local index and global index
+
+> **Global Index:** A global index is a one-to-many relationship, allowing one index partition to map to many table partitions. This means global index can be partitioned by the range or hash method, and it can be defined on any type of partitioned, or non-partitioned, table.
+>
+> **Local Index:** A local index is a one-to-one mapping between a index partition and a table partition, which means index is also partition by the same conditions of table.
+
+### Partitioned table index in TiDB
+
+In TiDB, index entries are encoding to key-value pairs, like: (In practice, index encoding is more complex, see [this article](https://pingcap.com/blog/2017-07-11-tidbinternal2/#map) and two proposals about [full collations](https://github.com/pingcap/tidb/pull/14574) and [cluster index](https://github.com/pingcap/tidb/blob/master/docs/design/2020-05-08-cluster-index.md) for more information.)
+
+```
+Key: tablePrefix{tableID}_indexPrefixSep{indexID}_indexedColumnsValue
+Value: rowID
+```
+
+Each partition of the partitioned table is treated as a  independent physical table in the TiKV layer, with its own `partitionID` . Therefore, the index entries of the partitioned table are encoded as:
+
+```
+Key: tablePrefix{partitionID}_indexPrefixSep{indexID}_indexedColumnsValue
+Value: rowID
+```
+
+So, in the current table partition implementation, a partition and its indexes are encoded to the same key-value range. Or we can  say, TiDB only has local index.
+
+This design is compatible with the MySQL, but also makes TiDB have the same [constraint](https://dev.mysql.com/doc/refman/5.7/en/partitioning-limitations-partitioning-keys-unique-keys.html) as MySQL, which says:
+
+> Every unique key on the table must use every column in the table's partitioning expression.
+
+For example, here is a partitioned table:
+
+```sql
+CREATE TABLE t (
+    A int,
+    B int,
+    UNIQUE KEY (B)
+)
+PARTITION BY RANGE (A) (
+ PARTITION p0 VALUES LESS THAN (10),
+ PARTITION p1 VALUES LESS THAN (20),
+ PARTITION p2 VALUES LESS THAN (maxvalue)
+);
+```
+
+Then we insert 2 records into it.
+
+```sql
+INSERT INTO t (A, B) VALUES (1, 23);
+INSERT INTO t (A, B) VALUES (11, 23);
+```
+
+The first record will be inserted into `p0`, the second will be inserted into `p1`. The index on column `B` will create two entries which keys are:
+
+```
+p0_idxB_23
+p1_idxB_23
+```
+
+Because index entries is built on different partition, uniqueness of index is broken. So, In MySQL and current TiDB, these indexes are not allowed.
+
+## Proposal
+
+### Overview
+
+This document proposes to support global index for partitioned tables. Global index use `tableID` instead of `partitionID`  as the prefix of  index entries. Since all entries in the partitioned table use same prefix, it can ensure index entries is unique in whole table, so that removing the constraint in partitioned tables indexes.
```

Comment:
```suggestion
This document proposes to support the global index for partitioned tables. Global index use `tableID` instead of `partitionID`  as the prefix of  index entries. Since all entries in a partitioned table use the same prefix, it can ensure index entry is unique in the whole table, thus removing the constraint in partitioned tables indexes.
```